### PR TITLE
prov/verbs: Fix additional klocwork issues

### PR DESCRIPTION
**Pull request by a-ilango**

### CODE REVIEW

File: prov/verbs/src/ep_rdm/verbs_tagged_ep_rdm_states.c [hunk 1]
```diff
@@ -547,6 +550,10 @@ fi_ibv_rdm_try_unexp_recv(struct fi_ibv_rdm_request *request,
 			if (request->parent) {
 				repost = fi_ibv_rdm_repost_multi_recv(request,
 					 found_request->len, rdata->ep);
+				if (!repost) {
+					ret = -FI_ENOMEM;
+					break;
+				}
 			}
```
Comment (shefty) [inline prov/verbs/src/ep_rdm/verbs_tagged_ep_rdm_states.c]: I don't think we want to abort here. We need to do something more with the unexpected message.

**Comment by shefty** [comment]
I have the same concerns on most of the changes.  I'm not sure that it's safe to simply abort with an error if we can't repost the buffer.  The code doesn't look structured to handle an error with reposting.  Maybe there's some way to avoid that error completely, possibly be re-using a structure that gets associated with the multi-recv buffer the first time it's posted?

**Comment by a-ilango** [comment]
But the error here is -FI_ENOMEM, isn't it okay to not worry about additional processing?

**Comment by shefty** [comment]
The enomem is that we can't repost a multi-recv buffer.  We still have the received data to deal with.  And if we can't repost the buffer, then the app won't get more receives, or know that the buffer is no longer usable, and basically everything just silently starts not working correctly...

**Comment by shefty** [comment]
I don't see an easy way out of any of this.  Please at least add a log warning if the repost fails.  AFAICT, we're basically doomed if that happens.

**Comment by a-ilango** [comment]
> The enomem is that we can't repost a multi-recv buffer

The ENOMEM is when util_buf_alloc fails. The buf pool for this is created with no limit on max memory. So in this specific case we would get enomem only when the system runs out of memory. So we might be okay aborting here.

I can add a warn log to let the user know. What do you think?

**Comment by shefty** [comment]
I think to handle an error, we would need to re-work the code a lot in this area.  The best we can do now is add a warning, plus the checks that you do have.

**Comment by a-ilango** [comment]
Agreed. I have added a warn log for util_buf_alloc failure.

**Pull request merged**